src/Stratis.Features.SQLiteWalletRepository/SQLiteWalletRepository.cs

```diff
@@ -872,12 +877,12 @@ private bool ParallelProcessBlock(ProcessBlocksInfo round, Block block, ChainedH
                         }
 
                         this.EndBatch(round);
+
+                        if (chainedHeader == null)
```

Comment:
Why was this check moved to be inside the `if`?

Reply:
Once of the conditions for entering the `if` is the condition being tested.

Reply:
I dont follow.. Previously, `if (round.NewTip != null || chainedHeader == null)` was resolved to false we would check the chainedHeader and if null, we would return false... with your change we don't?

Reply:
if `chainedHeader == null` we would return `false`, both before and after the change.

Reply:
are you sure there is no code path that leads to the call `this.StartBatch(round, chainedHeader)` in case chainedHeader is null?

because otherwise I see StartBatch should throw on L386
`round.PrevTip = (header.Previous == null) ? new HashHeightPair(0, -1) : new HashHeightPair(header.Previous);`

just watching code here on github, didn't had a chance to run it